src/object-store/src/property.hpp

```diff
@@ -55,6 +55,7 @@ namespace realm {
 
         size_t table_column = -1;
         bool requires_index() const { return is_primary || is_indexed; }
+        bool is_indexable() const { return type == PropertyTypeInt || type == PropertyTypeBool || type == PropertyTypeString || type == PropertyTypeDate; }
```

Comment:
If the plan is to migrate dates to double fields, then perhaps we shouldn't let them be indexed?